src/views/UserResourceDetails.vue

```diff
@@ -0,0 +1,212 @@
+      }
+    }
+
+    async function getCurrentResourceQuantityAndUser() {
+      const url = router.options.history.state.back; // /users/userId
+      const userId = url.substring(url.lastIndexOf("/") + 1); // substing to find userId
+      try {
+        await store.dispatch("users/fetchResourcesPerUser", userId);
+        const user = computed(() => store.getters["users/getUserResources"]);
+        const currentUserQuantity = user.value.resourcesAndQuantities;
+        selectedUser.value = user.value.owner.name;
+
+        for (const index in currentUserQuantity) {
+          const currentUserResource = currentUserQuantity[index].resource;
+          if (currentUserResource.id === resourceId) {
+            const resourceQuantity = currentUserQuantity[index].quantity;
+            quantity.value = resourceQuantity;
+            currentQuantity.value = resourceQuantity;
+          }
+        }
+      } catch (error) {
+        showErrorSnackbar("Couldn't get user and its resource/quantity");
+      }
+    }
+
+    if (route.path.includes("/add")) {
+      getCurrentResourceQuantity();
+    } else if (route.path.includes("/remove")) {
+      getCurrentResourceQuantityAndUser();
+    }
+
+    const handleAddSubmit = async () => {
+      const { valid } = await form.value.validate();
+      const findUser = allUsers.value.find(
+        (user) => user.name == selectedUser.value
+      );
+      if (valid) {
+        const data = {
+          userId: findUser.id,
+          resourceId: props.id,
+          quantity: quantity.value,
+        };
+        try {
+          await store.dispatch("users/postResourcePerUser", data);
+          showSuccessSnackbar("Successfully added quantity!");
+          router.push("/resources");
+        } catch (error) {
+          showErrorSnackbar("Couldn't not add quantity");
+        }
+      }
+    };
+
+    const handleRemoveSubmit = async () => {
+      // DELETE /resource/availability/{userId}/{resourceId}/{quantity}
+      const { valid } = await form.value.validate();
+      const findUser = allUsers.value.find(
+        (user) => user.name == selectedUser.value
+      );
+
+      if (valid) {
+        const data = {
+          userId: findUser.id,
+          resourceId: props.id,
+          quantityNumber: quantity.value,
+        };
+
+        try {
+          await store.dispatch("resources/removeQuantityFromResource", data);
+          router.push(`/users/${findUser.id}`);
+        } catch (error) {
+          showErrorSnackbar("Couldn't not update quantity");
+        }
+      }
+    };
+
+    const showErrorSnackbar = (message) => {
+      showSnackbar({
+        message: message,
+        color: "error",
+        timeout: 4000,
+        location: "bottom center",
+      });
+    };
+
+    const showSuccessSnackbar = (message) => {
+      showSnackbar({
+        message: message,
+        color: "success",
+        timeout: 4000,
+        location: "bottom center",
+      });
+    };
```

Comment:
I see this is copy-pasted from ResourceDetails. I suggest extracting this into the notification.js file.

Reply:
its dont think we can since we cannot inject a code that is in Vue component into a JS file